MaterialLib/PorousMedium/Permeability/Permeability.h

```diff
@@ -0,0 +1,103 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file:   Permeability.h
+ *
+ */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include "BaseLib/Error.h"
+#include "ProcessLib/Parameter/Parameter.h"
+
+namespace MaterialLib
+{
+namespace PorousMedium
+{
+/// The class implements a basic permeability model that employs a parameter
+/// (for instance a constant parameter or mesh cell dependend parameter) to fill
+/// the intrinsic permeability tensor.
+class Permeability
+{
+public:
+    explicit Permeability(
+        ProcessLib::Parameter<double> const& permeability_parameter,
+        int const dimension)
+        : _permeability_parameter(permeability_parameter), _dimension(dimension)
+    {
+        auto const number_of_components =
+            _permeability_parameter.getNumberOfComponents();
+
+        switch (number_of_components)
+        {
+            case 1:
+                if (_dimension != 1)
+                    OGS_FATAL(
+                        "The given parameter has one component, but the "
+                        "permeability tensor is defined for a %d dimensional "
+                        "problem.",
+                        _dimension);
+                break;
+            case 4:
+                if (_dimension != 2)
+                    OGS_FATAL(
+                        "The given parameter has 4 components, but the "
+                        "permeability tensor is defined for a %d dimensional "
+                        "problem.",
+                        _dimension);
+                break;
+            case 9:
+                if (_dimension != 3)
```

Comment:
all cases could be handled shorter when comparing `number_of_components` to `_dimension` squared.